flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdColumnUniqueness.scala

```diff
@@ -81,22 +77,11 @@ class FlinkRelMdColumnUniqueness private extends MetadataHandler[BuiltInMetadata
       return false
     }
 
-    // TODO get uniqueKeys from TableSchema of TableSource
-
-    relOptTable match {
-      case table: FlinkPreparingTableBase => {
-        val ukOptional = table.uniqueKeysSet
-        if (ukOptional.isPresent) {
-          if (ukOptional.get().isEmpty) {
-            false
-          } else {
-            ukOptional.get().exists(columns.contains)
-          }
-        } else {
-          null
-        }
-      }
-      case _ => rel.getTable.isKey(columns)
+    val uniqueKeys = FlinkRelMdUniqueKeys.INSTANCE.getTableUniqueKeys(relOptTable)
```

Comment:
I prefer we can get unique keys through public API: `mq.getUniqueKeys()` instead of depending on `FlinkRelMdUniqueKeys` class directly.